### PR TITLE
Add deploy action

### PR DESCRIPTION
**Pull request by zwolf**
This adds a simple action that uses reusable workflows located in the https://github.com/zooniverse/ci-cd repo to `npm build` (storing the result as artifact), then deploy to blob storage. 

PR https://github.com/zooniverse/ci-cd/pull/3 adds the workflows required for this and should be merged first.

### CODE REVIEW

**Review by eatyourgreens — APPROVED** [review]
LGTM. 

I think we can get rid of the build step. It builds a style sheet that’s already in the `public` directory. I don’t think it makes any changes to the existing CSS. 

The deploy script is written for a single page app, so might need to be changed to set short cache lifetimes for all `.html` files. I’m not sure. Given how infrequently this repo changes, a 1 week cache for content pages might be OK.

File: .github/workflows/deploy.yaml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    secrets:
+      creds: ${{ secrets.AZURE_STATIC_SITES }}
+
+  slack_notification:
+    name: Send Slack notification
+    uses: zooniverse/ci-cd/.github/workflows/slack_notification.yml@main
+    needs: deploy
+    if: always()
+    with:
+      deploy_url: 'https://data.galaxyzoo.org'
```
Comment (eatyourgreens) [inline .github/workflows/deploy.yaml:35]: Should this be `title_link`?

Comment (zwolf) [reply .github/workflows/deploy.yaml:35]: 👍 Yeah, copied that block from the PFE action, which will need to be updated when it switches over eventually

**Comment by zwolf** [comment]
Agreed about the build, but I mostly wanted to test drive the new workflows in the ci-cd repo and ensure that they properly cache/store/retrieve the build products before trying it elsewhere--so if you want, you can check those new workflows here: https://github.com/zooniverse/ci-cd/pull/3

If this looks good to you, I'll update this PR to upload directly to upload to `zooniverse-static/data.galaxyzoo.org` instead of `datatest` too.

**Comment by eatyourgreens** [comment]
The new workflows look good. The 1 week cache is the only thing that rings alarm bells for me, in case someone does change one of the HTML pages then wonders why they get stale content in the browser after deploying the change. 

It’s probably not a big deal but could the deploy script allow the 1 week max-age to be overridden by an input parameter? Then set it to say 10 minutes for this deploy? Just in case there is a problem we’ve not thought of.

**Comment by zwolf** [comment]
Nothing except the index.html has changed in years, but I hear you. I like the idea of making the TTLs configurable w/ defaults, if you think that will be used enough to warrant it.

**Comment by eatyourgreens** [comment]
I think all the other static sites are single pages with cache-busting file names for the CSS and JS files. This might be the only repo with multiple pages, so the only one that would use that parameter. And, like you said, those files haven’t changed in years. We could probably publish them with even longer cache lifetimes and no one would notice. 

I think I’m worrying about something that isn’t likely to be a real problem.

**Pull request merged**